### ISSUE TITLE
"pyoxidizer run" seems to fail if there are spaces in the build path (Windows)

### ISSUE THREAD

**Issue by dirtbirb** [indygreg/PyOxidizer#391]
Completely rewriting this issue because the original problem was dumb (just didn't have Rust installed -- though there were two issues there: the installer didn't put pyoxidizer on user or system path, and after I did that manually, it wasn't clear that the lack of Rust was the problem).

After fixing the above, I discovered that it still wouldn't build, with the relevant output at the bottom of this message. I moved the project to another folder with no spaces in the path and it worked great. Seems like it needs some quotes somewhere?

> building with Rust 1.51.0
> error: failed to run `rustc` to learn about target-specific information
> 
> Caused by:
>   process didn't exit successfully: `rustc - --crate-name ___ --print=file-names -LC:\Users\myusername\example dir name\example name\repo\test-suite\pyapp\.\build\python_distributions\python.ebf4acc5dff2\python\install --target x86_64-pc-windows-msvc --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg` (exit code: 1)
>   --- stderr
>   error: multiple input filenames provided (first two filenames are `-` and `dir`)
> 
> error[PYOXIDIZER_PYTHON_EXECUTABLE]: adding PythonExecutable to FileManifest
> 
> Caused by:
>     0: building Python executable
>     1: building executable with Rust project
>     2: command ["C:\\Users\\myusername\\.cargo\\bin\\cargo.EXE", "build", "--target", "x86_64-pc-windows-msvc", "--target-dir", "C:\\Users\\myusername\\AppData\\Local\\Temp\\pyoxidizerjEnDAe\\build\\target", "--bin", "pyapp", "--no-default-features", "--features", "build-mode-prebuilt-artifacts cpython-link-default"] exited with code 101
>    --> .\pyoxidizer.bzl:270:5
>     |
> 270 |     files.add_python_resource(".", exe)
>     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ to_file_manifest()
> 
> 
> error: adding PythonExecutable to FileManifest
> 
> Caused by:
>     0: building Python executable
>     1: building executable with Rust project
>     2: command ["C:\\Users\\myusername\\.cargo\\bin\\cargo.EXE", "build", "--target", "x86_64-pc-windows-msvc", "--target-dir", "C:\\Users\\myusername\\AppData\\Local\\Temp\\pyoxidizerjEnDAe\\build\\target", "--bin", "pyapp", "--no-default-features", "--features", "build-mode-prebuilt-artifacts cpython-link-default"] exited with code 101

**Comment by dirtbirb** [comment]
Seems related to #380, possibly also #390?

**Comment by dirtbirb** [comment]
Okay, now that I actually read the dependencies instead of just installing the .exe, I see I need Rust installed. My fault, will install now and verify that it works. 

I guess that could be more explicit in the error output itself, or mentioned in the installer or something.

**Comment by indygreg** [comment]
Yes, the Rust dependency has tripped many people up. The next release (hopefully out this week) automatically manages Rust for you so end-users don't have to install Rust as a separate dependency.

The policy of this project is the `main` branch is stable. So if you want to live on the edge, you can grab Windows installers from the GitHub Actions `release` workflow outputs. Although I wouldn't be surprised if there were subtle issues in the behavior of these builds.

Anyway, your new issue is quite interesting. This may be a bug in https://github.com/Kimundi/rustc-version-rs due to not handling paths cleanly on Windows. I'll have to dig into it before I point fingers though.

Regardless of the root cause, we should probably detect "bad" paths and fail fast with an actionable error message.
